tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -1605,24 +1604,182 @@ def _test_memory_based_scheduling_enabled_false(
+        constraint="ondemand1-i1",
+        other_options="--mem=4000 --test-only",
+        raise_on_error=False,
+    )
+    assert_that(result.stdout).is_equal_to("allocation failure: Requested node configuration is not available")
+
+    # Check that compatible nodes would be selected
+    result = slurm_commands.submit_command(
+        nodes=1,
+        command="sleep 1",
+        other_options="--mem=4000 --test-only",
+        raise_on_error=False,
+    )
+    assert_that(result.stdout).matches(r"^.*Job \d* to start.*$")
+    assert_that(result.stdout).does_not_contain("ondemand1-i1")
+
+    # Check that the `--mem` option only filters compute nodes instead of managing memory required by jobs
+    job_id_1 = slurm_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={
+            "nodes": 1,
+            "slots": 1,
+            "command": "sleep 30",
+            "other_options": "--mem=2000 -w queue1-st-ondemand1-i1-1",
+            "raise_on_error": False,
+        }
+    )
+    job_id_2 = slurm_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={
+            "nodes": 1,
+            "slots": 1,
+            "command": "sleep 30",
+            "other_options": "--mem=2000 -w queue1-st-ondemand1-i1-1",
+            "raise_on_error": False,
+        }
+    )
+    time.sleep(jiff)
+    assert_that(slurm_commands.get_job_info(job_id_1, field="JobState")).is_equal_to("RUNNING")
+    assert_that(slurm_commands.get_job_info(job_id_2, field="JobState")).is_equal_to("RUNNING")
+    # Here two jobs submitted with `--mem=2000` can fit on a node with less than 4000 MiB memory
+    # because without memory as consumable resource, Slurm doesn't track the memory usage of
+    # each job.
+    assert_that(slurm_commands.get_job_info(job_id_1, field="NodeList")).is_equal_to(
+        slurm_commands.get_job_info(job_id_1, field="NodeList")
+    )
+    slurm_commands.wait_job_completed(job_id_1)
+    slurm_commands.wait_job_completed(job_id_2)
+
+    # Check that without memory constraining, jobs might contend memory on the compute node
+    job_id_1 = slurm_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={
+            "nodes": 1,
+            "slots": 1,
+            "command": "srun ./a.out 2000000000",
+            "other_options": "--mem=2500 -w queue1-st-ondemand1-i1-1",
+            "raise_on_error": False,
+        }
+    )
+    time.sleep(jiff)
+    job_id_2 = slurm_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={
+            "nodes": 1,
+            "slots": 1,
+            "command": "srun ./a.out 2000000000",
+            "other_options": "--mem=2500 -w queue1-st-ondemand1-i1-1",
+            "raise_on_error": False,
+        }
+    )
+    slurm_commands.wait_job_completed(job_id_1)
+    slurm_commands.wait_job_completed(job_id_2)
+    # In this scenario the second job will have stolen memory from the first job, causing
```

Comment:
Why do we expect the second job to steal memory instead of having the first job acquiring it and the second one failing?

Reply:
That's what I've always seen happening. I would leave it as it is, and if someday we see these particular assertions failing, we will learn something new.